A debugging hack.

Check to make sure the dev_id passed to request_irq() and
free_irq() match.  Since the args are void pointers, it is
easy to pass the wrong variable.

Hacked-by: Geoff Levand <geoffrey.levand@am.sony.com>
---
 kernel/irq/manage.c |   15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

--- a/kernel/irq/manage.c
+++ b/kernel/irq/manage.c
@@ -13,6 +13,10 @@
 #include <linux/interrupt.h>
 #include <linux/slab.h>
 
+#ifdef CONFIG_PPC_PS3
+#include <asm/firmware.h>
+#endif
+
 #include "internals.h"
 
 #ifdef CONFIG_SMP
@@ -452,8 +456,17 @@ void free_irq(unsigned int irq, void *de
 			struct irqaction **pp = p;
 
 			p = &action->next;
-			if (action->dev_id != dev_id)
+			if (action->dev_id != dev_id) {
+#ifdef CONFIG_PPC_PS3
+			    if (firmware_has_feature(FW_FEATURE_PS3_LV1)) {
+				pr_debug("%s:%d: irq %u bad dev_id: request_irq(%p) != "
+					"free_irq(%p)\n" , __func__, __LINE__, irq, action->dev_id,
+					 dev_id);
+				BUG();
+			    }
+#endif
 				continue;
+			}
 
 			/* Found it - now remove it from the list of entries */
 			*pp = action->next;
